programs/drift/src/instructions/user.rs

```diff
@@ -318,6 +321,128 @@ pub fn handle_resize_swift_user_orders<'c: 'info, 'info>(
+        authority: ctx.accounts.authority.key(),
+        user_stats_fuel_insurance: user_stats.fuel_insurance,
+        user_stats_fuel_deposits: user_stats.fuel_deposits,
+        user_stats_fuel_borrows: user_stats.fuel_borrows,
+        user_stats_fuel_positions: user_stats.fuel_positions,
+        user_stats_fuel_taker: user_stats.fuel_taker,
+        user_stats_fuel_maker: user_stats.fuel_maker,
+        fuel_sweep_fuel_insurance: fuel_sweep.fuel_insurance,
+        fuel_sweep_fuel_deposits: fuel_sweep.fuel_deposits,
+        fuel_sweep_fuel_borrows: fuel_sweep.fuel_borrows,
+        fuel_sweep_fuel_positions: fuel_sweep.fuel_positions,
+        fuel_sweep_fuel_taker: fuel_sweep.fuel_taker,
+        fuel_sweep_fuel_maker: fuel_sweep.fuel_maker,
+    });
+
+    fuel_sweep.update_from_user_stats(&user_stats, clock.unix_timestamp.cast()?)?;
+    user_stats.reset_fuel();
+
+    Ok(())
+}
+
+pub fn handle_reset_fuel_season<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, ResetFuelSeason<'info>>,
+) -> Result<()> {
+    let mut user_stats = load_mut!(&ctx.accounts.user_stats)?;
+
+    let fuel_sweep = ctx.fuel_sweep();
+    user_stats.validate_fuel_sweep(&fuel_sweep)?;
+
+    let clock = Clock::get()?;
+    if let Some(fuel_sweep_account) = fuel_sweep {
+        // if FuelSweep exists, sweep before resetting user_stats
+        let mut fuel_sweep = load_mut!(fuel_sweep_account)?;
+        emit!(FuelSweepRecord {
+            ts: clock.unix_timestamp.cast()?,
+            authority: ctx.accounts.authority.key(),
+            user_stats_fuel_insurance: user_stats.fuel_insurance,
+            user_stats_fuel_deposits: user_stats.fuel_deposits,
+            user_stats_fuel_borrows: user_stats.fuel_borrows,
+            user_stats_fuel_positions: user_stats.fuel_positions,
+            user_stats_fuel_taker: user_stats.fuel_taker,
+            user_stats_fuel_maker: user_stats.fuel_maker,
+            fuel_sweep_fuel_insurance: fuel_sweep.fuel_insurance,
+            fuel_sweep_fuel_deposits: fuel_sweep.fuel_deposits,
+            fuel_sweep_fuel_borrows: fuel_sweep.fuel_borrows,
+            fuel_sweep_fuel_positions: fuel_sweep.fuel_positions,
+            fuel_sweep_fuel_taker: fuel_sweep.fuel_taker,
+            fuel_sweep_fuel_maker: fuel_sweep.fuel_maker,
+        });
+        fuel_sweep.update_from_user_stats(&user_stats, clock.unix_timestamp.cast()?)?;
+
+        emit!(FuelSeasonRecord {
+            ts: clock.unix_timestamp.cast()?,
+            authority: ctx.accounts.authority.key(),
+            fuel_insurance: fuel_sweep.fuel_insurance,
+            fuel_deposits: fuel_sweep.fuel_deposits,
+            fuel_borrows: fuel_sweep.fuel_borrows,
+            fuel_positions: fuel_sweep.fuel_positions,
+            fuel_taker: fuel_sweep.fuel_taker,
+            fuel_maker: fuel_sweep.fuel_maker,
+            fuel_total: fuel_sweep.total_fuel()?,
+        });
+        fuel_sweep.reset_fuel(clock.unix_timestamp.cast()?);
```

Comment:
i think its weird to reset the user stats but not the sweep account?

Reply:
this is resetting the fuel sweep account if it exists, userstats is reset below outside the `if` statement